zkevm-circuits/src/mpt_circuit/mod_extension.rs

```diff
@@ -0,0 +1,325 @@
+use eth_types::Field;
+use gadgets::util::Scalar;
+use halo2_proofs::plonk::{Error, VirtualCells};
+
+use super::{
+    helpers::{
+        ext_key_rlc_calc_value, KeyData, ListKeyGadget, ListKeyWitness, MPTConstraintBuilder,
+    },
+    rlp_gadgets::RLPItemWitness,
+    MPTContext,
+};
+use crate::{
+    circuit,
+    circuit_tools::{
+        cached_region::CachedRegion,
+        cell_manager::Cell,
+        constraint_builder::RLCChainableRev,
+        gadgets::{IsEqualGadget, LtGadget},
+    },
+    mpt_circuit::{
+        helpers::{ext_key_rlc_expr, Indexable, ParentData, FIXED, KECCAK},
+        param::HASH_WIDTH,
+        witness_row::StorageRowType,
+        FixedTableTag, RlpItemType,
+    },
+};
+
```

Comment:
Let me know if the doc makes sense
```suggestion
/// Validity conditions for ModExtensionGadget:
/// 1. Both the long and short extension nodes are in the parent node.
/// 2. Both the branches of the short extension node’s and the long extension node’s must be the same
/// 3. The concatenation of the middle extension node and short ext node must equal the long ext node
```